### PR TITLE
Re add pop sensitivity and path check

### PR DESCRIPTION
**Pull request by abdullahhamadeh**
I can't see my changes from #195 and #196.   I have merged them with develop again.

### CODE REVIEW

**Comment by Yuri05** [comment]
@abdullahhamadeh Tried this PR with [SensiCrash.zip](https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/files/4671544/SensiCrash.zip) and got the error message

```
[1] "23/05/2020 - 12:23:03 : Run sensitivity for simulation: Rifampin po, 450 mg, Bioequivalence - PopSim"
Error in data.frame(Outputs = outputColumn, pkParameters = pkParameterColumn,  :
  arguments imply differing number of rows: 85, 80
```

File: R/messages.R [hunk 1]
```diff
@@ -84,8 +84,12 @@ messages <- list(
...
     paste0("Workflow type '", workflowType, "' requires one reference population, but no reference population was provided.")
   },
-  
-  errorNotADimension = function(values){
+
+  errorNotADimension = function(values) {
     paste0("Expected a dimension. Check that '", paste0(values, collapse = "', '."), "' is included in ospsuite::allAvailableDimensions().")
+  },
+
+  invalidOuputPath = function(pth, simName) {
```
Comment (Yuri05) [inline R/messages.R:92]: typo: **pth**

File: R/utilities-sensitivity-analysis.R [hunk 2]
```diff
@@ -569,45 +571,68 @@ plotTornado <- function(data,
...
+                                      logFolder = NULL,
+                                      settings,
+                                      workflowType = NULL,
+                                      xParameters = NULL,
+                                      yParameters = NULL) {
+  dFList <- list()
+  i <- 0
+  for (structureSet in structureSets) {
+    indexDf <- read.csv(file = structureSet$popSensitivityAnalysisResultsIndexFile)
+    rankFilter <- 12
```
Comment (Yuri05) [inline R/utilities-sensitivity-analysis.R:591]: Still a TODO here. s. https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/issues/171#issuecomment-624548277

**Comment by abdullahhamadeh** [comment]
Fixes #171 
Fixes #167 
Fixes #165 
Fixes #164

**Comment by Yuri05** [comment]
@abdullahhamadeh With this PR I get the error message in plot sensitivity task for pop.
```
[1] "24/05/2020 - 14:58:42 : Starting: Plot Sensitivity task in Alpha Testing"
Fehler in rClr::clrCall(self$ref, "AllPKParameterSensitivitiesFor", pkParameterName,  : 
  Type:    System.MissingMethodException
Message: Could not find a suitable instance method AllPKParameterSensitivitiesFor on type OSPSuite.Core.Domain.SensitivityAnalyses.SensitivityAnalysisRunResult for method parameters String, String, SymbolicExpression
```
I use latest ospsuite-R package version **9.0.76**
[SensiCrash_2.zip](https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/files/4673557/SensiCrash_2.zip)

**Comment by abdullahhamadeh** [comment]
I will look at the mean sensitivity plot issue but for now this pr fixes the population sensitivity plot crash.  The cause of the crash was that the threshold wasn't being set to anything by default.  It had to be added explicitly.

I have fixed this so that we have the following way of setting the threshold upon initialization of the workfow: When the pop workflow is initialized it looks for whether a threshold was explicitly set for the pop sensitivity plot using the SensitivityPlotSettings class.  If not, it looks at the population sensitivity analysis settings.  If a set of perturbation parameter paths has been set there, the threshold is 1, else 0.9.  If the population sensitivity analysis settings are null, an instance of SensitivityPlotSettings class is used to set the theshold to 0.9 by default.

There is an issue currently, which is that if the sensitivity analysis is updated with a set of perturbation parameter paths _after_ the initialization of the workflow, the pop sensitivity plot task has no way of knowing that the threshold should be updated to 1 (instead of 0.9).  

Perhaps the simulation structure should also hold a copy of the task settings?  It's a situation where a pointer to the pertubation parameter paths would be extremely useful becuase then the threshold would automatically update depending on whether these paths are set.

**Comment by Yuri05** [comment]
oki, merging for now, let's create separate issues for the remaining topics

**Pull request merged**